yt_idefix/data_structures.py

```diff
@@ -885,6 +870,26 @@ def _parse_parameter_file(self):
             self.filename, geometry=self.parameters["definitions"]["geometry"]
         )
 
+        with h5py.File(self.filename, mode="r") as h5f:
+            root = list(h5f.keys())[0]
+            self._detected_field_list = list(h5f[f"{root}/vars/"])
+            self._field_name_map = {}
+            # some versions of Pluto define field names in lower case
+            # so we normalize standard output field names to upper case
+            # to avoid duplicating data in PlutoFields.known_other_fields
+            for varname in self._detected_field_list:
+                if (
+                    varname.upper()
+                    in dict(self._field_info_class.known_other_fields).keys()
```

Comment:
same as above.
To add to my previous comment, iterating over a dictionary itself is the same as iterating over its keys, so `.keys()` is redundant here